include/libhal-arm-mcu/stm32f1/timer.hpp

```diff
@@ -606,23 +575,30 @@ class advanced_timer_manager
    * @brief Creates a quadrature encoder object used to measure the rotation of
    * a motor that gives out quadrature encoder feedback.
    *
-   * @param encoder_pins The pins must be part of the same timer and they must
-   * correspond to channel 1 and 2 respectively. Any channel other than channel
-   * 1 and 2 will throw an hal::operation_not_permitted error. If the channels
-   * are flipped, it will result degrees counted in the wrong direction.
-   *
-   * @return hal::v5::strong_ptr<hal::rotation_sensor> object.
+   * @param p_allocator Provide the resource that this can be allocated to.
+   * @param p_pin1 First timer pin corresponding to channel 1
+   * @param p_pin2 Second timer pin corresponding to channel 2
+   * @param p_pulses_per_rotation Cycles per revolution for the quadrature
+   encoder.
+
+   * @return hal::v5::strong_ptr<hal::rotation_sensor>
    *
    * @throws hal::device_or_resource_busy - if timer is already being used with
    * a conflicting resource.
    * @throws hal::operation_not_permitted error - if any of the channels are not
    * 1 and 2.
    */
-  [[nodiscard]] hal::stm32f1::quadrature_encoder acquire_quadrature_encoder(
-    timer_pins p_pin1,
-    timer_pins p_pin2);
+  [[nodiscard]] hal::v5::strong_ptr<hal::rotation_sensor>
+  acquire_quadrature_encoder(std::pmr::polymorphic_allocator<> p_allocator,
+                             timer_pins p_pin1,
+                             timer_pins p_pin2,
+                             float p_pulses_per_rotation);
 
 private:
+  template<typename... T>
+  static hal::v5::strong_ptr<hal::rotation_sensor> create(
+    std::pmr::polymorphic_allocator<> p_allocator,
+    T&&... args);
```

Comment:
```suggestion
  template<typename... Args>
  static hal::v5::strong_ptr<hal::rotation_sensor> create(
    std::pmr::polymorphic_allocator<> p_allocator,
    Args&&... args);
```

This is me being very pedantic, but T... is something I rarely see and what I usually see is `Args...`. So if we could keep these consistent that would be great.